fury/ui/elements.py

```diff
@@ -3052,3 +3055,236 @@ def directory_click_callback(self, i_ren, _obj, listboxitem):
                 self.set_slot_colors()
         i_ren.force_render()
         i_ren.event.abort()
+
+
+class Shape2D(UI):
+    """Create and Manage 2D Shapes.
+    """
+
+    def __init__(self, shape_type, position=(0, 0)):
+        """Init this UI element.
+
+        Parameters
+        ----------
+        shape_type : string
+            Type of shape to be created.
+        position : (float, float)
+            (x, y) in pixels.
+        """
+        self.shape_type = shape_type
+        super(Shape2D, self).__init__(position)
+
+    def _setup(self):
+        """Setup this UI component.
+
+        Create a Shape.
+        """
+        if self.shape_type == "line":
+            self.shape = Rectangle2D(size=(2, 2))
+        elif self.shape_type == "quad":
+            self.shape = Rectangle2D(size=(2, 2))
+        elif self.shape_type == "circle":
+            self.shape = Disk2D(outer_radius=2)
+        else:
+            raise IOError("Unknown shape type: {}.".format(self.shape_type))
+
+        self.shape.on_left_mouse_button_pressed = self.left_button_pressed
+        self.shape.on_left_mouse_button_dragged = self.left_button_dragged
+
+    def _get_actors(self):
+        """Get the actors composing this UI component."""
+        return self.shape
+
+    def _add_to_scene(self, scene):
+        """Add all subcomponents or VTK props that compose this UI component.
+
+        Parameters
+        ----------
+        scene : scene
+
+        """
+        self.shape.add_to_scene(scene)
+
+    def _get_size(self):
+        return self.shape.size
+
+    def _set_position(self, coords):
+        """Set the lower-left corner position of this UI component.
+
+        Parameters
+        ----------
+        coords: (float, float)
+            Absolute pixel coordinates (x, y).
+        """
+        if self.shape_type == "circle":
+            self.shape.center = coords
+        else:
+            self.shape.position = coords
+
+    def rotate_point(self, x, y, deg):
+        new_x = np.cos(deg) * x - y * np.sin(deg)
+        new_y = np.sin(deg) * x + y * np.cos(deg)
+        return new_x, new_y
+
+    def rotate_line(self, deg):
+        line_actor = self.shape
+
+        for i in range(4):
+            current_x, current_y = line_actor._points.GetPoint(i)[:2]
+            new_x, new_y = self.rotate_point(current_x, current_y, deg)
+            line_actor._points.SetPoint(i, new_x, new_y, 0.0)
+
+        line_actor._polygonPolyData.SetPoints(line_actor._points)
+        mapper = PolyDataMapper2D()
+        mapper = set_input(mapper, line_actor._polygonPolyData)
+
+        line_actor.actor.SetMapper(mapper)
+
+    def resize(self, size):
+        """Resize the UI.
+        """
+        if self.shape_type == "line":
+            hyp = np.hypot(size[0], size[1]) * (-1 if size[0] < 0 else 1)
+            self.shape.resize((hyp, 2))
+            self.rotate_line(deg=np.arctan(size[1]/size[0]))
```

Comment:
I encountered a few RuntimeErrors in this line:
```
RuntimeWarning: invalid value encountered in long_scalars
  self.rotate_line(deg=np.arctan(size[1]/size[0]))
```
Also, I faced the same RuntimeError due to division by 0.

Reply:
Solved 👍